src/blockdata/block.rs

```diff
@@ -122,6 +127,13 @@ impl Block {
 }
 
 impl BlockHeader {
+    /// Return the block hash.
+    pub fn block_hash(&self) -> BlockHash {
+        let mut engine = BlockHash::engine();
+        self.consensus_encode(&mut engine).expect("engines don't error");
+        BlockHash::from_engine(engine)
+    }
```

Comment:
I'm interested why not we use `BlockHash::hash` function instead and use longer version with `HashEngine`

Reply:
Because to use `BlockHash::hash` you first need to allocate the serialized block into a `Vec<u8>`. It's more efficient to serialize the data directly into the engine so you don't need to allocate a byte vector that can be up to 4 MiB in size.

Reply:
Ok, see it. Then better to re-factor this repeated code into a different function – I wrote this suggestion elsewhere: https://github.com/rust-bitcoin/rust-bitcoin/pull/391#discussion_r366243967

Reply:
I don't really think this 3 lines of code are a problem, though.